discord/sink.py

```diff
@@ -0,0 +1,775 @@
+        reception reports from other receivers to measure round-trip
+        propagation to those receivers.
+    rts: :class:`int`
+        RTP timestamp. Corresponds to the same time as the NTP timestamp (above),
+        but in the same units and with the same random offset as the RTP
+        timestamps in data packets.
+    spc: :class:`int`
+        The total number of RTP data packets transmitted by the sender since
+        starting transmission up until the time this SR packet was generated.
+        The count is reset if the sender changes its SSRC identifier.
+    soc: :class:`int`
+        The total number of payload octets (i.e., not including header or padding)
+        transmitted in RTP data packets by the sender since starting transmission
+        up until the time this SR packet was generated. The count is reset if
+        the sender changes its SSRC identifier.
+    report_blocks: Sequence[:class:`RTCPReceiverReport`]
+        Sequence of :class:`RTCPReceiverReport` objects that tell statistics.
+        Receivers do not carry over statistics when a source changes its SSRC
+        identifier due to a collision.
+    extension: :class:`bytes`
+        Profile-specific extension that may or may not contain a value.
+    """
+    __slots__ = RTCPPacket.__slots__ + (
+        "ssrc", "nts", "rts", "spc", "soc", "report_blocks",
+        "extension"
+    )
+
+    def __init__(self, version_flag: int, rtcp_type: RTCPMessageType, length: int, data: bytes):
+        super().__init__(version_flag, rtcp_type, length)
+
+        fmt = ">IQ3I"
+        buf_size = struct.calcsize(fmt)
+        self.ssrc, self.nts, self.rts, self.spc, self.soc = \
+            struct.unpack(fmt, data[:buf_size])
+
+        self._parse_report_and_extension(data[buf_size:])
+
+
+class RTCPReceiverReportPacket(RTCPPacket):
+    """RTCP Receiver Report packet which provides quality feedback.
+
+    Read in detail here: https://www.freesoft.org/CIE/RFC/1889/20.htm
+
+    Extends :class:`RTCPPacket` and inherits its attributes.
+
+    Attributes
+    ----------
+    ssrc: :class:`int`
+        The synchronization source identifier for the originator of this SR packet.
+    report_blocks: Sequence[:class:`RTCPReceiverReport`]
+        Sequence of :class:`RTCPReceiverReport` objects that tell statistics.
+        Receivers do not carry over statistics when a source changes its SSRC
+        identifier due to a collision.
+    extension: :class:`bytes`
+        Profile-specific extension that may or may not contain a value.
+    """
+    __slots__ = RTCPPacket.__slots__ + (
+        "ssrc", "report_blocks", "extension"
+    )
+
+    def __init__(self, version_flag: int, rtcp_type: RTCPMessageType, length: int, data: bytes):
+        super().__init__(version_flag, rtcp_type, length)
+
+        fmt = ">I"
+        buf_size = struct.calcsize(fmt)
+        self.ssrc = struct.unpack(fmt, data[:buf_size])[0]
+        self._parse_report_and_extension(data[buf_size:])
```

Comment:
- Should probably prefer `!` to `>` (same byte order, but `!` has a specific meaning as to being network order)
- Shouldn't need to call calcsize on a literal
- Shouldn't slice a buffer passed to unpack, use unpack_from
- this last function should probably be removed rather than just exist with code commented out.


```suggestion
        self.ssrc = struct.unpack_from("!I", data, 0)[0]
        self.report_blocks = []
        self.data = data[4:]
```

Reply:
> this last function should probably be removed rather than just exist with code commented out.

I've kept it there since it would be used if discord sent valid rtcp packets. 
All the other points are fair. I've checked your other reviews as well, so I'll go through and make the changes.